### PR TITLE
[server] don't error on project not_found

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

We need to handle the not_found error during when projects are actually optional, i.e. during workspace start.

## Related Issue(s)

Fixes WEB-628

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - se-start-wa387281c97</li>
	<li><b>🔗 URL</b> - <a href="https://se-start-wa387281c97.preview.gitpod-dev.com/workspaces" target="_blank">se-start-wa387281c97.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - se-start-ws-wo-project-gha.14480</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Review by geropl — APPROVED** [review]
Code LGTM :white_check_mark: 

@svenefftinge Do you want me to do some testing as well or did you already do so?

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -1593,9 +1607,6 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         const user = await this.checkAndBlockUser("getPrebuildEvents");
 
         const project = await this.projectsService.getProject(user.id, projectId);
-        if (!project) {
```
Comment (svenefftinge) [inline components/server/src/workspace/gitpod-server-impl.ts:1596]: Removed these unnecessary checks, as getProject already throws that error

File: components/server/src/workspace/workspace-starter.ts [hunk 2]
```diff
@@ -696,10 +696,7 @@ export class WorkspaceStarter {
         metadata.setMetaId(workspace.id);
         if (workspace.projectId) {
             metadata.setProject(workspace.projectId);
-            const project = await this.projectDB.findProjectById(workspace.projectId);
-            if (project && project.teamId) {
-                metadata.setTeam(project.teamId);
-            }
+            metadata.setTeam(workspace.organizationId);
```
Comment (svenefftinge) [inline components/server/src/workspace/workspace-starter.ts:699]: no need to fetch the project as we have the org id on the workspace

**Comment by svenefftinge** [comment]
I did not, but will do when the preview env is up

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 3]
```diff
@@ -844,17 +844,6 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         await this.userDeletionService.deleteUser(user.id);
     }
 
-    private async getTeamMembersByProject(projectId: string | undefined): Promise<TeamMemberInfo[]> {
```
Comment (svenefftinge) [inline components/server/src/workspace/gitpod-server-impl.ts:847]: this was unnecessariuly going through project. We now have an organizationId on workspaces so I replaced these with `organizationService.listMembers`.

**Pull request merged**